content/algorithms/gossip_sub.md

```diff
@@ -1,21 +1,92 @@
 ---
 title: "GossipSub"
 weight: 6
-dashboardWeight: 1.5
-dashboardState: incomplete
-dashboardAudit: 0
+dashboardWeight: 2.0
+dashboardState: stable
+dashboardAudit: 1
 dashboardTests: 0
 ---
 
 # GossipSub
 ---
 
-Messages and block headers along side the message references are propagated using the [gossipsub libp2p pubsub router](https://github.com/libp2p/specs/tree/master/pubsub/gossipsub). Every full node must implement and run that protocol. All pubsub messages are authenticated and must be [syntactically validated](message#message-syntax-validation) before being propagated further.
+Transaction messages and block headers alongside the message references are propagated using the [libp2p GossipSub router](https://github.com/libp2p/specs/tree/master/pubsub/gossipsub). **Both miners and full nodes must implement and run this protocol.** All pubsub messages are authenticated and must be [syntactically validated](message#message-syntax-validation) before being propagated further. The GossipSub Specification ++ provides all the implementation details and parameter settings. Here we provide an overview of the main characteristics and design choices of GossipSub.
 
-Further more, every full node must implement and offer the bitswap protocol and provide all Cid Referenced objects, it knows of, through it. This allows any node to fetch missing pieces (e.g. `Message`) from any node it is connected to. However, the node should fan out these requests to multiple nodes and not bombard any single node with too many requests at a time. A node may implement throttling and DDoS protection to prevent such a bombardment.
+GossipSub is a gossip-based pubsub protocol that is utilising two types of links to propagate messages: i) _mesh links_ that carry full messages in an _eager-push_ manner and ii) _gossip-links_ that carry message identifiers only and realise a _lazy-pull_ propagation model.
 
```

Comment:
Everything below this line is copy & paste from the Gossipsub spec. I propose that we link to it rather than forcing us to maintain it in two places.

Reply:
I totally agree!

Reply:
I don't think all this detail belongs in the spec; we have a (very) comprehensive spec and a paper, we shouldn't be repeating ourselves.
So yeah, delete and link to stuff.